c-api/compat-5.3.c

```diff
@@ -336,23 +383,124 @@ COMPAT53_API void luaL_traceback (lua_State *L, lua_State *L1,
 
 
 COMPAT53_API int luaL_fileresult (lua_State *L, int stat, const char *fname) {
+  const char* s = NULL;
   int en = errno;  /* calls to Lua API may change this value */
   if (stat) {
     lua_pushboolean(L, 1);
     return 1;
   }
   else {
     lua_pushnil(L);
+#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (defined(__GNUC__) && (__GNUC__ < 6) && defined (__MINGW32__))
```

Comment:
`(defined(__GNUC__) && (__GNUC__ < 6) && defined (__MINGW32__)` shouldn't `strerror_r` also be used when `__GNUC__ >= 6` ?

Reply:
My patch is only for MinGW 5.x. I have not thoroughly tested 6 or 7, but it is said to have `strerror_r` in the implementation because this code never errored on my 6 or 7 MinGW versions when I ran the tests otherwise. If someone can show me MinGW 6 or 7 having it (there were patches submitted to MinGW implementing `strerror_r`, but I don't know if they were accepted), I can change the conditional.

Reply:
Your 'patch' uses strerror, so why don't you just remove '(defined(__GNUC__) && (__GNUC__ < 6) && defined (__MINGW32__)' altogether, and let it fallthrough to strerror here?

Reply:
Because the maybe-unsafe version is a last resort, to avoid tipping off linters, hyper-pedantic warning/error checkers, and others.

Reply:
On mingw it uses strerror either way though right now, so your custom implementation doesn't make any difference.